service/matching/matching_engine.go

```diff
@@ -911,10 +913,16 @@ func (e *matchingEngineImpl) UpdateWorkerVersioningRules(
 				req.GetDeleteCompatibleRedirectRule(),
 			)
 		case *workflowservice.UpdateWorkerVersioningRulesRequest_CommitBuildId_:
+			hasRecentPoller := false
+			if ptqm, err := tqMgr.GetVersionedPhysicalTaskQueueManager(ctx, req.GetCommitBuildId().GetTargetBuildId()); ptqm != nil && err == nil {
+				hasRecentPoller = ptqm.HasPollerAfter(time.Now().Add(-versioningPollerSeenWindow))
```

Comment:
I'd prefer to keep versioned queue business private to partition manager and instead of GetVersionedPhysicalTaskQueueManager, expose a the following methods:
- `HasPollerAfter(buildId, time)` empty being unversioned
- `HasAnyPollerAfter(time)` for checking all versioned and unversioned.